src/main/java/duke/storage/LocalStorage.java

```diff
@@ -0,0 +1,143 @@
+package duke.storage;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidArgumentException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Local storage to store the data from local file.
+ */
+public class LocalStorage {
+    private File file;
+
+    /**
+     * Constructor to create a local storage using the data from local file.
+     * @param path
+     */
+    public LocalStorage(String path) {
+        File localFile = readFile(path);
+        this.file = localFile;
+    }
+
+    /**
+     * Read from file in local duke.storage.
+     * @param path Path to the file in the local duke.storage.
+     * @return File obtained from local storage
+     */
+    public static File readFile(String path) {
+        File file = new File(path);
+        file.getParentFile().mkdirs();
+
+        if (file.exists()) {
+            return file;
+        }
+
+        try {
+            file.createNewFile();
+        } catch (IOException io_error) {
+            io_error.printStackTrace();
```

Comment:
Might not be good for the stack trace to be printed like this. Maybe you could reformat the error message to be nicer for the user to read? Otherwise, its good that it is verbose and the user knows that there is an error there